src/ast.rs

```diff
@@ -346,7 +392,43 @@ pub enum BinaryOpKind {
     BinaryAnd,
     BinaryOr,
 }
-
+impl Compile for BinaryOpKind {
+     fn compile(&self, t: &Target) -> String {
+        match self {
+            BinaryOpKind::Add => return String::from("+"),
+            BinaryOpKind::Sub => return String::from("-"),
+            BinaryOpKind::Mul => return String::from("*"),
+            BinaryOpKind::Div => return String::from("/"),
+            BinaryOpKind::Assign => return String::from("EQU"),
+            BinaryOpKind::Eq => return String::from("=="),
+            BinaryOpKind::Ne => return String::from("NEQ"),
+            BinaryOpKind::Gt => return String::from("GTR"),
+            BinaryOpKind::Lt => return String::from("LSS"),
+            BinaryOpKind::GtEq => return String::from("GEQ"),
+            BinaryOpKind::LtEq => return String::from("LEQ"),
+            BinaryOpKind::Shr => return String::from(">>"),
+            BinaryOpKind::Shl => return String::from("<<"),
+            BinaryOpKind::Xor => return String::from("^"),
+            BinaryOpKind::LogicalAnd => return String::from("&&"),
+            BinaryOpKind::LogicalOr => return String::from("||"),
+            BinaryOpKind::BinaryAnd => return String::from("&"),
+            BinaryOpKind::BinaryOr => return String::from("|"),
+            _ => return String::from("unexpected symbol type"),
```

Comment:
You shouldn't need an underscore case here since its an enum. you should be matching on all variants